eth/transaction_receipt.go

```diff
@@ -30,3 +38,125 @@ func (t *TransactionReceipt) TransactionType() int64 {
 
 	return t.Type.Int64()
 }
+
+// RequiredFields inspects the Transaction Type and returns an error if any required fields are missing
+func (t *TransactionReceipt) RequiredFields() error {
+	var fields []string
+	switch t.TransactionType() {
+	case TransactionTypeLegacy:
+		// LegacyReceipt is rlp([status, cumulativeGasUsed, logsBloom, logs])
+		// only .Status is a pointer at the moment
+		// NOTE: pre-EIP-658 receipts include root and not status: rlp([root, cumulativeGasUsed, logsBloom, logs])
+		if t.Root != nil && t.Status != nil {
+			return fmt.Errorf("receipt contains both root and status")
+		}
+		if t.Root == nil && t.Status == nil {
+			fields = append(fields, "one of status or root")
+		}
+		return nil
+	case TransactionTypeAccessList:
+		// The EIP-2718 ReceiptPayload for this transaction is rlp([status, cumulativeGasUsed, logsBloom, logs]).
+		// Same as TransactionTypeLegacy.
+		if t.Status == nil {
+			fields = append(fields, "status")
```

Comment:
i see `fields` is appended to here & also for `TransactionTypeLegacy` (line 54) but nil is returned unconditionally, is that intentional?